src/tel_download.js

```diff
@@ -345,38 +379,35 @@
   // For webk /k/ webapp
   setInterval(() => {
     /* Voice Message */
-    const voiceMessages = document.querySelectorAll("audio-element");
+    const pinnedAudio = document.body.querySelector(".pinned-audio");
+    let dataMid;
+    let downloadButtonPinnedAudio = document.body.querySelector("._tel_download_button_pinned_container") || document.createElement("button");
+    if (pinnedAudio) {
+      dataMid = pinnedAudio.getAttribute("data-mid");
+      downloadButtonPinnedAudio.className = "btn-icon tgico-download _tel_download_button_pinned_container";
+      downloadButtonPinnedAudio.innerHTML =
+          '<span class="tgico button-icon">\uE93D</span>';
+    }
+    const voiceMessages = document.body.querySelectorAll("audio-element");
     voiceMessages.forEach((voiceMessage) => {
-      if (voiceMessage.querySelector(".audio-waveform") === null) {
-        return; /* Skip non-voice message */
-      }
-      if (voiceMessage.querySelector("_tel_download_button_voice_container")) {
+      const bubble = voiceMessage.closest(".bubble");
+      if (!bubble || bubble.querySelector("._tel_download_button_voice_container")) {
         return; /* Skip if there's already a download button */
       }
+      if (dataMid && downloadButtonPinnedAudio.getAttribute("data-mid") !== dataMid && voiceMessage.getAttribute("data-mid") === dataMid) {
+        downloadButtonPinnedAudio.onclick = (e) => {
+          e.stopPropagation();
+          tel_download_audio(link);
+        }
+        downloadButtonPinnedAudio.setAttribute("data-mid", dataMid);
+      }
       const link = voiceMessage.audio && voiceMessage.audio.getAttribute("src");
       if (link) {
-        const container = document.createElement("div");
-        container.className = "_tel_download_button_voice_container";
-        container.style.position = "absolute";
-        container.style.width = "100%";
-        container.style.height = "100%";
-        container.style.display = "flex";
-        container.style.justifyContent = "center";
-        container.style.alignItems = "end";
-        const downloadButton = document.createElement("button");
-        downloadButton.className =
-          "btn-icon default__button tgico-download tel-download";
-        downloadButton.style.marginBottom = "16px";
-        downloadButton.style.backgroundColor = "black";
-        downloadButton.onclick = (e) => {
-          e.stopPropagation();
-          tel_download_audio(voiceMessage.audio.getAttribute("src"));
-        };
-        voiceMessage.closest(".bubble").appendChild(container);
-        container.appendChild(downloadButton);
+        pinnedAudio.querySelector(".pinned-container-wrapper-utils").appendChild(downloadButtonPinnedAudio);
```

Comment:
here, the button will constantly updating its innerHTML for every tick even though nothing has changed on the page, and thus the button is kinda "blinking". Could you update the logic here to avoid update if the button is already there and nothing is required to update?

```suggestion
  if (link && downloadButtonPinnedAudio.parentElement !== pinnedAudio.querySelector(".pinned-container-wrapper-utils"))
                pinnedAudio.querySelector(".pinned-container-wrapper-utils").appendChild(downloadButtonPinnedAudio);
```

![2023-09-26 12 21 21](https://github.com/Neet-Nestor/Telegram-Media-Downloader/assets/23090573/e0237be0-2fb6-4ed1-acfa-45b6452b0f51)